nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/image/ResizeImage.java

```diff
@@ -196,4 +187,53 @@ public void process(final InputStream rawIn, final OutputStream out) throws IOEx
         }
     }
 
+    // https://stackoverflow.com/questions/7951290/re-sizing-an-image-without-losing-quality
```

Comment:
it is polite to cite the source.  however, stackoverflow posts cannot be copied as their origin/licensing is not established or clear.  The code to perform the function you desire needs to be original/authentic or sourced to ALv2 or category-A compliant code https://www.apache.org/legal/resolved.html#category-a

We cannot accept this code as-is.

Reply:
@joewitt  StackOverflow has the following disclaimer:

> site design / logo © 2018 Stack Exchange Inc; user contributions licensed under cc by-sa 3.0 with attribution required. rev 2018.7.5.30950

Reply:
So yeah, it's possible that the code was stolen from somewhere else, but the prima facie licensing of that content is CC share alike. From what I read on the licensing page, it's only CC non commercial that is problematic.

Reply:
Please see my research notes below. Also updated with correct original source URL, and fixed checkstyle issues.